### ISSUE TITLE
Notifications not showing

### ISSUE THREAD

**Issue by shaaw** [bastimeyer/livestreamer-twitch-gui#224]
Just like the tittle says.
I have windows 10 with every update there is available, notifications enabled in the App. Other apps show notifications, but i dont get any notification ever. Some time ago they were working.
I also have a laptop with same OS and same version of livestreamer-twitch-gui and there the notifications work.

**Comment by bastimeyer** [comment]
Did you disable notifications for Livestreamer Twitch GUI inside Windows?
Go to `Settings -> System -> Notifications & actions` and see if you've disabled it...
http://www.tomshardware.co.uk/faq/id-2569603/control-individual-app-notifications-windows.html

**Comment by shaaw** [comment]
No, its Activated in windows.

**Comment by bastimeyer** [comment]
Please start the application with the `--remote-debugging-port=8888` parameter, open `http://localhost:8888` in your local webbrowser, click on the `Livestreamer Twitch GUI` link and see if there are any errors being printed in the console when a new channel has come online (the GUI queries the Twitch API once every minute).

**Comment by shaaw** [comment]
There was no error when i did this. But somehow i fixed it. Instead of extracting new versions en the same folder and replacing the files, i deleted the folder and created a new one with the new version and now is working

**Comment by Poorchop** [comment]
I'm having this issue too. I just formatted my PC and did a fresh install of Windows 8.1, and notifications are only showing up as balloons over the tray icon, whereas they used to show up in the upper right corner of the screen. I checked my notification settings and I see a bunch of applications are enabled, but Livestreamer Twitch GUI is not one of them. Another one of my third party applications is showing up there as well and it's properly showing desktop notifications, so it's only this application that seems to be having issues.

**Comment by bastimeyer** [comment]
@Poorchop Windows requires a startmenu shortcut for applications to be able to create these toast notifications instead of balloon tooltips. Did you disable the shortcut setting?
https://github.com/bastimeyer/livestreamer-twitch-gui/blob/v0.13.0/src/app/models/localstorage/Settings.js#L60
Advanced settings need to be enabled, though.
https://github.com/bastimeyer/livestreamer-twitch-gui/blob/v0.13.0/src/templates/Settings.hbs#L150-L153
As you can see, the GUI will always try to re-create the shortcut on startup or when re-enabling the setting.
https://github.com/bastimeyer/livestreamer-twitch-gui/blob/v0.13.0/src/app/nwjs/setup/shortcut.js#L59-L66

**Comment by Poorchop** [comment]
Yes I enabled the advanced settings and I just tried unchecking and rechecking the box for the start menu shortcut, then I relaunched the application, but I still don't see the start menu shortcut. I have another node-webkit application installed that successfully created a start menu shortcut however, so I'm not sure why it's not working for this application. The only major change from my previous Windows installation is that I was using Classic Shell, but I haven't reinstalled it yet on my current installation. I don't think that matters at all though.

I even deleted the old folder and tried extracting the archive again, but no luck. It has been a long while since I've tried debugging but if you point me in the right direction again, I could give it a try. In the meantime, I'll try the instructions that you posted above and wait for another stream to come online. I remember having issues trying to debug in Firefox but hopefully I'll see something this time.

**Comment by Poorchop** [comment]
I tried the debug method above but I didn't see any output on the page or in the web console. I've restarted my PC multiple times, checked the notification settings, tried creating new "installations" by extracting the archive again, and I tried debugging in both Firefox and IE. I still can't think of any reason as to why I'm suddenly having this issue after a fresh installation but I guess the start menu shortcut just isn't being created.

**Comment by bastimeyer** [comment]
Does NW.js try to create the shortcut? Set a breakpoint here:
https://github.com/bastimeyer/livestreamer-twitch-gui/blob/v0.13.0/src/app/nwjs/setup/shortcut.js#L41

That's the path where the start menu shortcut should be:
[%APPDATA%\Microsoft\Windows\Start Menu\Programs](https://github.com/bastimeyer/livestreamer-twitch-gui/blob/v0.13.0/package.json#L114) ([documentation](https://github.com/nwjs/nw.js/wiki/Notification#toast-notification-requires-node-webkit--v011))

If it does execute the NW.js method, then it's most like likely an issue with NW.js itself. Did you try removing the second NW.js app you are using?

**Comment by Poorchop** [comment]
All right, I set the breakpoint and it turns out that `createStartmenuShortcutWin8()` is never even getting called. It looks like the issue is somewhere [here](https://github.com/bastimeyer/livestreamer-twitch-gui/blob/3c5accdab7e00ec86aaa7beca4c0ef13d8f21140/src/app/nwjs/setup/shortcut.js#L59). I tried to dig a little further in the dev tools but debugging Ember is not my forte since it turns everything into a super class and adds a ton of methods to the `settings` object.

I checked %APPDATA%\..\Programs just in case and as expected, the shortcut is not there.

>Did you try removing the second NW.js app you are using?
As in completely uninstalling it? I can try uninstalling it to test but I've always had it installed alongside livestreamer-twitch-gui, and I never had problems in the past. It's [turtl](https://turtl.it/) if you're interested in testing on your system. I'm running 8.1 x64 with the 64-bit version of both applications.
